src/plots/gl3d/scene.js

```diff
@@ -480,9 +480,12 @@ proto.plot = function(sceneData, fullLayout, layout) {
             var axLetter = axis._name.charAt(0);
 
             for(j = 0; j < objects.length; j++) {
-                var objBounds = objects[j].bounds;
-                sceneBounds[0][i] = Math.min(sceneBounds[0][i], objBounds[0][i] / dataScale[i]);
-                sceneBounds[1][i] = Math.max(sceneBounds[1][i], objBounds[1][i] / dataScale[i]);
+                var obj = objects[j];
+                var objBounds = obj.bounds;
+                // add worse-case pad for cone traces
+                var pad = (obj._trace.data._normMax || 0) * dataScale[i];
```

Comment:
This is what I came up with to solve the "cones out of scene" problem.

```js
Plotly.newPlot(gd, [{
	type: 'cone',
      "x": [1, 2, 3],
      "y": [1, 2, 3],
      "z": [1, 2, 3],
      "u": [1, 0, 0],
      "v": [0, 3, 0],
      "w": [0, 0, 2],
}])
```

now gives:

![image](https://user-images.githubusercontent.com/6675409/40155290-d679fb88-5960-11e8-8ebb-49af1b19c4bb.png)

Reply:
Looks good! I suppose based on https://github.com/gl-vis/gl-cone3d/issues/6 we could drop it to 3/4 of that value, though then if we make that configurable it would depend on the offset... more trouble than it's worth, let's keep what you have here.